src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -863,7 +863,13 @@ private void DoCloseHelper()
                 }
 
                 if ((hostRunspace == null) || (this == hostRunspace))
-                {
+                {                   
+                    // We should close transcripting only if we are running in default runspace.
+                    if (executionContext.CurrentRunspace != null && DefaultRunspace != null && executionContext.CurrentRunspace.Id != DefaultRunspace.Id)
```

Comment:
Also I just realized that the "AmsiUtils.Uninitialize()" call should be done at the last runspace being closed as well.  Can you please move that call into this logic too?